### PR TITLE
Add CI action to sync published versions simple example to sandbox repository

### PR DESCRIPTION
**Pull request by djhi**
## Problem

As explained in #10252, Stackblitz is currently very slow. Besides, we can't use codesandbox to open a subpath of our monorepo anymore.

## Solution

We now have a dedicated repository for the simple example: https://github.com/marmelab/react-admin-sandbox
This PR adds a CI action that sync the monorepo simple example with the sandbox repository.
It does so only when a new tag is released, except for `next`, `beta` and `alpha` tags

## Additional Checks

- [x] The PR targets `master` for a bugfix, or `next` for a feature

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -157,7 +162,7 @@ jobs:
     runs-on: ubuntu-latest
     name: GreenFrame
     needs: [e-commerce]
-    if: success() && github.ref == 'refs/heads/master'
+    if: success() && github.ref == 'refs/heads/master' && github.ref_type != 'tag'
```
Comment (slax57) [inline .github/workflows/test.yml:165]: **thought**: on the contrary, we could run the Greenframe analysis only when a new tag/version is released, as opposed to any commit on `master` which might be a little too often. Wdyt?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -208,3 +213,21 @@ jobs:
...
+    # Only run on new tags that target a release (not latest nor next) and avoid alpha and beta tags
+    if: github.event_name == 'push' && github.ref_type == 'tag' && github.ref_name == 'refs/tags/v*' && !contains('beta', github.ref_name) && !contains('alpha', github.ref_name)
+    needs: [typecheck, simple-example-typecheck, unit-test, e2e-test]
+    steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+        - name: Use Node.js LTS
+          uses: actions/setup-node@v3
+          with:
+              node-version: '16.x'
```
Comment (slax57) [inline .github/workflows/test.yml:228]:
```suggestion
              node-version: '18.x'
```

To be consistent with other jobs of the workflow

File: Makefile [hunk 3]
```diff
@@ -172,3 +172,6 @@ storybook: ## Launch the storybook
 build-storybook: ## Build the storybook
 	@echo "Building storybook..."
 	@yarn build-storybook
+
+update-sandbox: ## Push the local version of the simple example to the sandbox repository
+	./update-demo.sh
```
Comment (slax57) [inline Makefile:177]:
```suggestion
	./update-sandbox.sh
```

(there is no `update-demo.sh`)

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -208,3 +213,21 @@ jobs:
...
+        - name: Update Sandbox Repository
+          env:
+              SSH_DEPLOY_KEY: ${{ secrets.SSH_SANDBOX_DEPLOY_KEY }}
```
Comment (slax57) [inline .github/workflows/test.yml:231]:
```suggestion
              SSH_SANDBOX_DEPLOY_KEY: ${{ secrets.SSH_SANDBOX_DEPLOY_KEY }}
```

To be consistent with the actual script

**Review by slax57 — CHANGES_REQUESTED** [review]

File: scripts/update-sandbox.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+(rm -rf $TEMPD/src)
+(rm -rf $TEMPD/assets)
+
+# Copy the demo files into the temporary directory
+(cp -r ./examples/simple/. $TEMPD)
+
+# Update the demo repository
+# Allow empty commits because even though we may have published new version of the enterprise packages,
+# the demo code may not have changed. However we still want it to be redeployed and, as we don't keep its
+# yarn.lock, it should use the new packages
```
Comment (slax57) [inline scripts/update-sandbox.sh:33]: I understand the rationale, but I believe @fzaninotto was in favor of actually computing and committing the `yarn.lock` file, in order to reduce the sandbox startup time

File: .github/workflows/test.yml (hunk 1 above)
Comment (djhi) [reply .github/workflows/test.yml:165]: @fzaninotto any opinion?

Comment (fzaninotto) [reply .github/workflows/test.yml:165]: I agree that we should only test greenframe when releasing a new version

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -208,3 +213,21 @@ jobs:
...
+          uses: actions/checkout@v3
+        - name: Use Node.js LTS
+          uses: actions/setup-node@v3
+          with:
+              node-version: '18.x'
+        - name: Update Sandbox Repository
+          env:
+              SSH_SANDBOX_DEPLOY_KEY: ${{ secrets.SSH_SANDBOX_DEPLOY_KEY }}
+              SANDBOX_REPOSITORY: ${{ secrets.SANDBOX_REPOSITORY }}
+          run: make update-sandbox
```
Comment (erwanMarmelab) [inline .github/workflows/test.yml:233]: **nitpick (non-blocking):** ‏Not sure that we will use this Makefile command manually. We may could directly run the script here and remove the new Makefile command.

File: scripts/update-sandbox.sh [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+(rm -rf $TEMPD/src)
+(rm -rf $TEMPD/assets)
+
+# Copy the demo files into the temporary directory
+(cp -r ./examples/simple/. $TEMPD)
+# Install dependencies to that the sandbox has a lock file
+yarn install --cwd $TEMPD
+
+# Update the demo repository
+(cd $TEMPD && git add -A && git commit --allow-empty -m "Update sandbox" && git push)
```
Comment (erwanMarmelab) [inline scripts/update-sandbox.sh:33]: **question:** ‏Why concatenate all those commands while the rest of the script has one command per line?

Comment (erwanMarmelab) [reply scripts/update-sandbox.sh:33]:
```suggestion
(cd $TEMPD)
(git add -A)
(git commit --allow-empty -m "Update sandbox")
(git push)
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (djhi) [reply .github/workflows/test.yml]: Done

File: scripts/update-sandbox.sh (hunk 7 above)
Comment (djhi) [reply scripts/update-sandbox.sh:33]: No reason and not important

File: .github/workflows/test.yml (hunk 6 above)
Comment (djhi) [reply .github/workflows/test.yml:233]: Useful for manual testing

**Review by erwanMarmelab — APPROVED** [review]

**Review by slax57 — APPROVED** [review]

**Pull request merged**